frontend/src/app/modules/calendars/calendars-page/calendars-page.component.ts

```
@@ -1,30 +1,148 @@
+
+    public connectAppleCalendar = new EventEmitter<void>();
+
+    constructor(
+        private teamService: TeamService,
+        private dialog: ConfirmationWindowService,
+        private calendarService: CalendarsService,
+        private notificationService: NotificationService,
+    ) {
+        super();
+        this.connectGoogleCalendar.subscribe(() => {
+            this.connectGoogle();
+        });
+    }
 
-        this.userCalendars = getDefaultCalendars();
+    ngOnInit(): void {
+        this.refreshData();
 
         this.checkOptions = getDefaultOptions();
     }
 
+    refreshData() {
+        this.teamService
+            .getCurrentUserTeams()
+            .pipe(this.untilThis)
+            .subscribe((response) => {
+                this.allTeams = response;
+            });
+
+        this.calendarService
+            .getUserGoogleCalendars()
+            .pipe(this.untilThis)
+            .subscribe((response) => {
+                this.userCalendars = response;
+
+                this.userCalendars.forEach((element) => {
+                    const teamsNames = element?.visibleForTeams?.map((x) => x.name);
+
+                    element.visibleForTeams = this.allTeams?.filter((x) => teamsNames?.includes(x.name));
+                    element.importEventsFromTeam = this.allTeams?.find(
+                        (x) => x.name === element.importEventsFromTeam?.name,
+                    );
+                });
+            });
+    }
+
+    removeCalendar(id: bigint) {
+        this.calendarService
+            .deleteGoogleCalendar(id)
+            .pipe(this.untilThis)
+            .subscribe((response) => {
+                if (response) {
+                    this.notificationService.showSuccessMessage('Calendar was successfully deleted');
+                } else {
+                    this.notificationService.showErrorMessage("Something went wrong. Calendar did't deleted");
+                }
+                this.refreshData();
```

it would be better to throw an error on the backend and handle it here